src/coreclr/nativeaot/Runtime/arm64/AsmMacros.h

```diff
@@ -245,6 +245,49 @@ __SECTIONREL_tls_CurrentThread SETS "$__SECTIONREL_tls_CurrentThread":CC:"_"
 
     MEND
 
+;; -----------------------------------------------------------------------------
+;;
+;; Macro to get a pointer to the tls_InlinedThreadStatics for the currently executing thread
+;;
+
+    GBLS __SECTIONREL_tls_InlinedThreadStatics
+__SECTIONREL_tls_InlinedThreadStatics SETS "SECTIONREL_tls_InlinedThreadStatics"
+
+    MACRO
+        INLINE_GET_INL_THREADSTATICS $destReg, $trashReg
```

Comment:
Is it difficult to have `INLINE_GET_TLS_VAR` on Windows Arm64 for some reason?

Reply:
Win Arm64 TLS seems more complex than x64. I am not sure if it needs to be like this.
It looks like it is hardcoding some mangled names while x64 does not:
```
    GBLS __SECTIONREL_tls_InlinedThreadStatics
__SECTIONREL_tls_InlinedThreadStatics SETS "SECTIONREL_tls_InlinedThreadStatics"
```
I think it still can be parameterized as a macro and then we would have only two uses - `_tls_CurrentThread` and `_tls_InlinedThreadStatics` and the second is temporary.

The main reason I chose to duplicate the code as it felt less error prone. I do not have a good way to test on win-arm64.

Ill revisit it to see what it would take to make this a macro. (and perhaps make it simpler)

Also since `RhpGetInlinedThreadStaticBase` helper is an optional optimization. We could just omit it on win-arm64.

Reply:
IIRC, there was a bug or limitation in the arm64 assembler that makes it impossible to inline the reloc into the instruction, and that's why it has to be split into two macros.

Reply:
Turns out it is possible to do something similar to x64, but the asm code must be very precise. It is more challenging because:  
- on arm64 we need two relocs - HI and LO. Easier to deviate from the pattern.
- arm64 instruction set allows semantical overlap of instructions when immediate offsets/shifts are trivial - i.e. `#0`  is optional, but omitting it may result in a different instruction. It would work the same, but the desired reloc may not be applicable and masm will either produce error, or more likely, silently ignore the reloc.
(i.e `add $destReg, $destReg, #0, lsl #0`  with `#0`s omitted could be a trivial move with nothing relocatable)
- the documentation is a bit misleading. For example
<html>
<body>
<!--StartFragment-->

IMAGE_REL_ARM64_SECREL_HIGH12A | 0x000A | Bit 12:23 of section offset of the target, for instructions ADD/ADDS (immediate) with zero shift.
-- | -- | --


<!--EndFragment-->
</body>
</html>
 
Is not completely correct. There _must_ be a 12bit shift for the HI reloc to work.

I can see how masm treatment of this pattern looked buggy/limited (or it could indeed be buggy in the past) and a more roundabout, but working pattern was adopted.